group/assignment6/RearrangeCars.cpp

```diff
@@ -3,155 +3,218 @@
 RearrangeCars::RearrangeCars(const vector<int> &startStates) : startStates(startStates) {}
 
 void RearrangeCars::setStartStates(const vector<int> &startStates) {
-	this->startStates = startStates;
+      this->startStates = startStates;
 }
 
 int RearrangeCars::findEmptySlotId() {
-	for (int slotId = 0; slotId < this->startStates.size(); slotId++) {
-		if (this->startStates[slotId] == -1) 
-			return slotId;		// no car has -1 id, it is reserved for empty slot
-	}
+      for (int slotId = 0; slotId < this->startStates.size(); slotId++) {
+            if (this->startStates[slotId] == -1)
+                  return slotId;                // no car has -1 id, it is reserved for empty slot
+      }
 
-	return -1;	// This code should not be reached
+      return -1;        // This code should not be reached
 }
 
 vector<int> RearrangeCars::findCarPositions(const vector<int> &currentStates) {
-	vector<int> carPositions(currentStates.size() - 1);
-	for (int i = 0; i < currentStates.size(); i++) {
-		if (currentStates[i] != -1)		// check for empty slot
-			carPositions[currentStates[i]] = i;
-	}
+      vector<int> carPositions(currentStates.size() - 1);
+      for (int i = 0; i < currentStates.size(); i++) {
+            if (currentStates[i] != -1)                // check for empty slot
+                  carPositions[currentStates[i]] = i;
+      }
 
-	return carPositions;
+      return carPositions;
 }
 
 void RearrangeCars::moveCar(vector<int> &currentStates, vector<int> &carPosition, int from, int to) {
-	currentStates[to] = currentStates[from];
-	currentStates[from] = -1;
-	carPosition[currentStates[to]] = to;
+      currentStates[to] = currentStates[from];
+      currentStates[from] = -1;
+      carPosition[currentStates[to]] = to;
 }
 
-bool RearrangeCars::validMove(int carId, int emptySlotId, const vector<set<int>>* constrains) {
-	if (!constrains)	// Constrains are not set
-		return true;
+bool RearrangeCars::validMove(int carId, int emptySlotId, const vector<set<int>> *constrains) {
+      if (!constrains)        // Constrains are not set

```

Comment:
I think, these are constrainTs, not constrains - we should fix this typo everywhere.

Reply:
I will correct this :)